addons/website/models/ir_model.py

```diff
@@ -38,3 +39,15 @@ def get_website_meta(self):
         # dummy version of 'get_website_meta' above; this is a graceful fallback
         # for models that don't inherit from 'website.seo.metadata'
         return {}
+
+    def _get_base_lang(self):
+        """ Returns the default language of the website as the base language if the record is bound to it """
+        self.ensure_one()
+        website = False
+        if 'website_id' in self._fields:
+            website = self.website_id
```

Comment:
Having a field called `website_id` does not tell what this many2one field stands for.
Interpreting that as "oh, this record must be bound to some website" is arbitrary and not even correct, as far as I could see.